go-controller/pkg/mock/mock_ovn.go

```diff
@@ -0,0 +1,469 @@
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get NAT List by Logical Router
+func (mock *MockOVNClient) LRNATList(lr string) ([]*goovn.NAT, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Add Meter with a Meter Band
+func (mock *MockOVNClient) MeterAdd(name, action string, rate int, unit string, external_ids map[string]string, burst int) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Deletes meters
+func (mock *MockOVNClient) MeterDel(name ...string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// List Meters
+func (mock *MockOVNClient) MeterList() ([]*goovn.Meter, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// List Meter Bands
+func (mock *MockOVNClient) MeterBandsList() ([]*goovn.MeterBand, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Add chassis with given name
+func (mock *MockOVNClient) ChassisAdd(name string, hostname string, etype []string, ip string, external_ids map[string]string,
+	transport_zones []string, vtep_lswitches []string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Delete chassis with given name
+func (mock *MockOVNClient) ChassisDel(chName string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get chassis by hostname or name
+func (mock *MockOVNClient) ChassisGet(chname string) ([]*goovn.Chassis, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get encaps by chassis name
+func (mock *MockOVNClient) EncapList(chname string) ([]*goovn.Encap, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Set NB_Global table options
+func (mock *MockOVNClient) NBGlobalSetOptions(options map[string]string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get NB_Global table options
+func (mock *MockOVNClient) NBGlobalGetOptions() (map[string]string, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Set SB_Global table options
+func (mock *MockOVNClient) SBGlobalSetOptions(options map[string]string) (util.OVNCommandInterface, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Get SB_Global table options
+func (mock *MockOVNClient) SBGlobalGetOptions() (map[string]string, error) {
+	return nil, fmt.Errorf("method %s is not implemented yet", functionName())
+}
+
+// Close connection to OVN
+func (mock *MockOVNClient) Close() error {
```

Comment:
If the connection is closed, further calls should fail. So we should track that state.

Reply:
Mimicking behavior of underlying library here, I am changing this to cause only `Execute` to fail. The commands can be still created.